cmd/client-fs.go

```diff
@@ -316,59 +306,34 @@ func (f *fsClient) put(reader io.Reader, size int64, metadata map[string][]strin
 	if len(metadata[metaDataKey]) != 0 {
 		attr, e = parseAttribute(metadata[metaDataKey][0])
 		if e != nil {
+			tmpFile.Close()
 			return 0, probe.NewError(e)
 		}
-		if err := preserveAttributes(partFile, attr); err != nil {
+		if err := preserveAttributes(tmpFile, attr); err != nil {
+			tmpFile.Close()
 			return 0, err.Trace(objectPath)
 		}
 	}
 
-	// Get stat to get the current size.
-	partSt, e := os.Stat(objectPartPath)
-	if e != nil {
-		err := f.toClientError(e, objectPartPath)
-		return 0, err.Trace(objectPartPath)
-	}
-
-	var totalWritten int64
-	// Current file offset.
-	var currentOffset = partSt.Size()
-
-	reader = hookreader.NewHook(reader, progress)
-
-	if !avoidResumeUpload && currentOffset > 0 {
-		// At this stage reader is always a seeker
-		// since it is a hook reader.
-		seeker, _ := reader.(io.Seeker)
-		if _, e = seeker.Seek(currentOffset, 0); e != nil {
-			return 0, probe.NewError(e)
-		}
-		// Discard bytes until currentOffset.
-		if _, e = io.CopyN(ioutil.Discard, progress, currentOffset); e != nil {
-			return 0, probe.NewError(e)
-		}
-	}
-
-	n, e := io.Copy(partFile, reader)
+	totalWritten, e := io.Copy(tmpFile, hookreader.NewHook(reader, progress))
 	if e != nil {
+		tmpFile.Close()
 		return 0, probe.NewError(e)
 	}
 
-	// Save currently copied total into totalWritten.
-	totalWritten = n + currentOffset
-
 	// Close the input reader as well, if possible.
 	closer, ok := reader.(io.Closer)
 	if ok {
 		if e = closer.Close(); e != nil {
+			tmpFile.Close()
 			return totalWritten, probe.NewError(e)
 		}
 	}
 
-	// Close the file before rename.
-	if e = partFile.Close(); e != nil {
-		return totalWritten, probe.NewError(e)
-	}
+	// Close the file before renaming, we need to do this
+	// specifically for windows users - windows explicitly
+	// disallows renames on Open() fd's by default.
+	tmpFile.Close()
```

Comment:
Let's return an error only in this tmpFile.Close() (easier to debug when one user complains)

Reply:
@vadmeste why are we bothered about error from Close() in which scenario?